course1/13.popular_package/8.package_json/task1.13.8.3/main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+type User struct {
+	Name     string    `json:"name"`
+	Age      int       `json:"age"`
+	Comments []Comment `json:"comments"`
+}
+
+type Comment struct {
+	Text string `json:"text"`
+}
+
+func writeJSON(filePath string, data []User) error {
+	writeData, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	dir := filepath.Dir(filePath)
+	err = os.MkdirAll(dir, 0755)
+	if err != nil {
+		return err
+	}
+
+	file, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if cerr := file.Close(); cerr != nil {
+			fmt.Printf("Error closing file %s: %s\n", filePath, cerr)
+		}
+	}()
+
+	_, err = file.Write(writeData)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func main() {
+	data := []User{
+		{
+			"Alice",
```

Comment:
set field names if you create a multi-line struct literal
you can omit field names If you set a one line literal
`User{"Alice", 56, "comment1"}`